reenact/static/js/charts.js

```diff
@@ -1,39 +1,158 @@
+      data: xaxis_labels,
+      axisLabel: { show: true, align: 'center' },
+      axisTick: { show: false }
+    },
+    yAxis: {
+      type: 'value',
+      splitLine: { show: false },
+      axisLabel: { show: false }
+    },
+    series: seriesList,
+    legend: {
+      right: '5%',
+      orient: 'vertical',
+      icon: 'circle',
+      textStyle: { fontSize: 12 },
+      tooltip: { show: true, formatter: legendTooltipFormatter }
+    }
+  };
+
+  return option;
 }
 
-function create_main_chart() {
+function update_chart_from_sliders() {
     const formData = new FormData(capacityForm);
     const params = new URLSearchParams(formData).toString();
-    get_echart_options("main_chart", params).then(
-        chartOptions => {createChart("mainChart", chartOptions);},
-    );
+    let requestUrl = window.location.origin + '/chart/main_chart?' + params;
+
+    fetch(requestUrl, {
+        method: 'GET',
+        mode: 'cors',
+        headers: new Headers({
+            'Accept': 'application/json',
+            'Content-Type': 'text/plain'
+        }),
+        credentials: 'same-origin',
+    })
+    .then(response => response.json())
+    .then(data => {
+        let options = generate_main_chart(data.production, data.demand);
+        createChart("mainChart", options);
+    })
+    .catch(error => console.error("Fehler beim Laden der Chart-Daten:", error));
 }
 
 function create_main_chart_on_startup() {
-  const productionDemandChartOptions = JSON.parse(document.getElementById("productionDemandChart").textContent);
-  createChart("mainChart", productionDemandChartOptions);
+        let requestUrl = window.location.origin + '/chart/main_chart';
+
+    fetch(requestUrl, {
+        method: 'GET',
+        mode: 'cors',
+        headers: new Headers({
+            'Accept': 'application/json',
+            'Content-Type': 'text/plain'
+        }),
+        credentials: 'same-origin'
+    })
+    .then(response => response.json())
+    .then(data => {
+        let options = generate_main_chart(data.production, data.demand);
+        createChart("mainChart", options);
+    })
+    .catch(error => console.error("Fehler beim Laden der Chart-Daten beim Startup:", error));
 }
 
+create_main_chart_on_startup();
+
+function reload_main_chart(scenarioNumber) {
+    let params = new URLSearchParams({ scenario: scenarioNumber }).toString();
+    let requestUrl = window.location.origin + '/chart/main_chart?' + params;
+
+    fetch(requestUrl, {
+        method: 'GET',
+        mode: 'cors',
+        headers: new Headers({
+            'Accept': 'application/json',
+            'Content-Type': 'text/plain'
+        }),
+        credentials: 'same-origin',
+    })
+    .then(response => response.json())
+    .then(data => {
+        let options = generate_main_chart(data.production, data.demand);
+        createChart("mainChart", options);
+    })
+    .catch(error => console.error("Fehler beim Laden der Szenario-Daten:", error));
+}
+
+
```

Comment:
These functions should be refactored into one function as they are quite similar. 
Startup chart could be integrated as first scenario. 
Parameters should be given as (optional?) input to the function and either be empty or fetched from form before calling the function.
Only error message has to be unified then.